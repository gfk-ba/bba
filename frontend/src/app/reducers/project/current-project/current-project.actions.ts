import { ApiDeleteRequest, ApiGetRequest, ApiPostRequest, ApiPutRequest } from '@dcs/ngx-tools';
import { Action } from '@ngrx/store';
import { currentProjectManager } from './current-project.manager';
import { IProject, Project } from '../models/project.class';

export class FetchCurrentProject extends ApiGetRequest {
  constructor(id: string) {
    super(`projects/${id}`, currentProjectManager.actions.fetch.base, currentProjectManager.schema);
  }
}

export class CreateCurrentProject extends ApiPostRequest {
  constructor(project: IProject) {
    super(
      `projects`,
      currentProjectManager.actions.create.base,
      project,
      currentProjectManager.schema
    );
  }
}

export class UpdateCurrentProject extends ApiPutRequest {
  constructor(project: Project) {
    super(
      `projects/${project.id}`,
      currentProjectManager.actions.update.base,
      project.toObject(),
      currentProjectManager.schema
    );
  }
}

export class DeleteCurrentProject extends ApiDeleteRequest {
  constructor(project: Project) {
    super(`projects/${project.id}`, currentProjectManager.actions.delete.base, project.toObject());
  }
}

export class ResetCurrentProject implements Action {
  readonly type = currentProjectManager.actions.fetch.reset;
}
