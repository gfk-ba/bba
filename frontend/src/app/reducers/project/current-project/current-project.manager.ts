import { schema } from 'normalizr';
import { State } from '../../index';
import { IProject, Project } from '../models/project.class';
import {
  crudEntityManagerFactory,
  INormalizedEntityState,
  generateNormalizedEntityState,
} from '@dcs/ngx-tools';

export interface ICurrentProjectState extends INormalizedEntityState {
  entities: { projects: { [key: string]: IProject } };
}

const initialState = generateNormalizedEntityState<ICurrentProjectState>({ projects: {} });

const projectSchema = new schema.Entity(
  'projects',
  {},
  {
    idAttribute(entity) {
      return String(entity.id);
    },
  }
);

export const currentProjectStateSelector = (state: State): ICurrentProjectState =>
  state.projects.currentProject;

export const currentProjectManager = crudEntityManagerFactory(
  'Current Project',
  initialState,
  currentProjectStateSelector,
  projectSchema,
  Project
);
