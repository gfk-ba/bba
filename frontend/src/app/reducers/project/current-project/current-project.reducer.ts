import { Action } from '@ngrx/store';
import { currentProjectManager, ICurrentProjectState } from './current-project.manager';

export const currentProject = (
  state: ICurrentProjectState = currentProjectManager.initialState,
  action: Action
): ICurrentProjectState => {
  switch (
    action.type
    // overwrite or add any actions here, just default redux
  ) {
  }

  return currentProjectManager.reducer(state, action);
};
