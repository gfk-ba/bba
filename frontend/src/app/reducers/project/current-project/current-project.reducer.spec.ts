import { currentProject } from './current-project.reducer';

describe('currentProject reducer', () => {
  it('works', () => {
    expect(currentProject(undefined, { type: 'any' })).toMatchSnapshot();
  });
});
