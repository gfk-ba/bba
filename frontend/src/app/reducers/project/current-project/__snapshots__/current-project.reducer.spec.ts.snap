// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currentProject reducer works 1`] = `
Object {
  "entities": Object {
    "projects": Object {},
  },
  "error": null,
  "lastState": null,
  "loaded": false,
  "loading": false,
  "result": "",
  "updatedAt": null,
  "updating": false,
}
`;
